Dockerfile

```diff
@@ -13,17 +13,24 @@ ENV LD_PRELOAD /usr/lib64/libgomp.so.1:/opt/intel/mkl/lib/intel64/libmkl_def.so:
 # Install necessary build tools
 RUN yum install -y gcc-c++ make swig3
 
-# Install necesary headers/libs
-RUN yum install -y python-devel numpy
+# Install Python and necesary headers/libs
+RUN yum -y install https://centos7.iuscommunity.org/ius-release.rpm && \
+    yum -y install python36u-devel python36u-pip && \
+    ln -sf /usr/bin/python3.6 /usr/bin/python && \
+    ln -sf /usr/bin/pip3.6 /usr/bin/pip && \
+    pip install -U pip numpy==1.17.0 scipy==1.3.1 && \
+    rm -rf /root/.cache/pip
```

Comment:
This seems overkill. The reason for tests failing here is that `scipy` is missing, so I'd be in favor of a fix that just adds (a suitable version of) scipy.

Reply:
I think upgrading to Python 3 is also a good idea because Python 2 [will retire in ~4 months](https://pythonclock.org/). Is not it?